packages/server/src/app/db/content/content.service.ts

```diff
@@ -111,6 +103,121 @@ export class ContentService {
         return await this.contentModel.updateOne({'_id': contentId, 'author': user.sub}, {'audit.isDeleted': true});
     }
 
+    /* Sections for Prose, Poetry, and Scripts */
+
+    /**
+     * Adds a section to some content. Only applicable to Prose, Poetry, and Scripts.
+     * 
+     * @param user The author of the content
+     * @param contentId The content ID
+     * @param sectionInfo The new section's info
+     */
+    async createSection(user: JwtPayload, contentId: string, sectionInfo: SectionForm) {
+        const work = await this.contentModel.findOne({'_id': contentId, 'author': user.sub, 'audit.isDeleted': false}, {autopopulate: false});
+
+        if (isNullOrUndefined(work)) {
+            throw new UnauthorizedException(`You don't have permission to do that.`);
+        } else {
+            return await this.sectionsService.createNewSection(sectionInfo).then(async sec => {
+                await this.contentModel.updateOne({'_id': contentId, 'author': user.sub, 'audit.isDeleted': false}, {$push: {'sections': sec._id}}, {strict: false});
+
+                return sec;
+            });
```

Comment:
you could simplify this a bit by just saying:
```suggestion
            const sec = await this.sectionsService.createNewSection(sectionInfo);
            await this.contentModel.updateOne({'_id': contentId, 'author': user.sub, 'audit.isDeleted': false}, {$push: {'sections': sec._id}}, {strict: false});

            return sec;
```